Spatial.gd

```diff
@@ -47,6 +47,15 @@ var taco_parser: TacoParser
 var x11_window_id_burrito: int
 var is_transient:bool = false
 
+# Variables that store informations about ui scaling
+# The ui-size as read from the link can have the values [0=small; 1=normal; 2=large; 3=larger]
+var ui_size = 1
+# This array holds the width of one item for every ui-scale
+const icon_size_preset = [26.5, 29.5, 33.0, 36.0] # 0=small; 1=normal; 2=large; 3=larger
```

Comment:
When we were trying to calculate the position before we found it tricky. We found the best position value for normal to be 323. I think we should just store the full position value instead of trying to calculate it. If down the line, GW2 adds another button, we can manually change the values.

Reply:
I am not sure with this one, my values might not be optimal but technically there is no big difference.
If I would change my value for `size: normal` to 29.36 we would end with 322.96 for our button.

If ANET adds another button we need to change one constant, if we do everything absolute we would need to change margin_left and margin_right four times each.
Doing it my way would also allow us to let the user change the position with a setting like, move button one step to the right / one step to the left.

That said, if you think it's not worth I will rewrite it.

Reply:
I'm inclined to agree with @klingbolt that it would be preferred over what is in this pr now. But there is probably an even better solution if you want to put in the legwork.

Something about having fractional values here sets off some alarms form me, at least for 'normal', I could maybe see an argument for why they would exist for small, maybe.

It is possible that the position of the icon is also not as simple as just a width, it could be there is an additional padding value somewhere before/after/between each icon, or a fixed left padding offset for just the first one. By finding the icon assets and lining them up against screenshots we can figure out what the exact coordinates are for each icon and then use those offsets to determine the correct algorithm. We can ignore all of that if we just used a fixed value.

Like you say, the downside to a single fixed value is if anet ever adds another button. Looking at the history of anet adding new buttons to gw2, I think this event will be *very* rare, and editing 4 numbers is not too much more work.

Reply:
Ok, i still have a different opinion but i am happy to implement the requested changes.
That means we want see something like this, right?
```
const button_margin {
	0: {"left": 292, "right": 318},
	1: {"left": 323, "right": 352},
	2: {"left": 363, "right": 396},
	3: {"left": 396, "right": 432}
}

$Control/GlobalMenuButton.margin_left = button_margin[self.ui_size]["left"]
$Control/GlobalMenuButton.margin_right = button_margin[self.ui_size]["right"]
```